docs/run_locally.md

```diff
@@ -0,0 +1,192 @@
+### How to install and use SeeChange on a local machine (not dockerized)
+
+As always, checkout the code from github: <https://github.com/c3-time-domain/SeeChange>.
+We recommend using a virtual environment to install the dependencies. For example:
+
+```bash
+python3 -m venv venv
+```
+
+Then activate the virtual environment and install the dependencies:
+
+```bash
+cd SeeChange
+source venv/bin/activate
+pip install -r requirements.txt
+```
+
+This covers the basic python dependencies. 
+Now we need to install postgres and set it up 
+to run on the default port (5432) with a database called `seechange`.
+
+On a mac, you can do this with homebrew:
+
+```bash
+brew install postgresql
+brew services start postgresql
+/usr/local/opt/postgres/bin/createdb seechange
+```
+
+Usually you will want to add the default user:
+    
+```bash
+/usr/local/opt/postgres/bin/createuser -s postgres
+```
+
+On linux/debian use
+ 
+```bash
+sudo apt install postgresql
+```
+
+Make sure the default port is 5432 in /etc/postgresql/14/main/postgresql.conf
+(assuming the version of postgres is 14).
+To restart the service do 
+
+```bash
+sudo service postgresql restart
+```
+
+
+To login to postgres (as the user "postgres"): 
+```bash
+sudo -u postgres psql
+```
+
+From here you can create or drop the database:
+
+```sql
+CREATE DATABASE seechange;
+DROP DATABASE seechange WITH(force);
+```
+
+To use the database, login as above but then change into the database:
+
+```sql
+\c seechange
+```
+
+
+#### Installing Q3C extension for postgres
+
+Get the code from <https://github.com/segasai/q3c>. 
+Installing following the instructions: 
+
+```bash
+make
+make install
+```
+
+Login to psql and do:
+
+```sql
+\c seechange
+CREATE EXTENSION q3c;
+```
+
+#### Getting the database schema up-to-date
+
+The database schema is managed by alembic.
+If the database is in a fresh state (just created), do:
+
+```bash
+alembic upgrade head
+```
+
+If the database has already been used (e.g., on a differnt branch), you may need to do:
+
+```bash
+alembic downgrade base
+alembic upgrade head
+```
+
+To generate a new migration script, do:
+
+```bash
+alembic revision --autogenerate -m "message"
+```
+
+#### Installing submodules
+
+To install submodules used by SeeChange 
+(e.g., the `nersc-upload-connector` package that is used to connect the archive)
+do the following:
+
+```bash
+cd extern/nersc-upload-connector
+git submodule init
+```
+
+If those packages require updates, you can do that from the root SeeChange directory
+using:
+
+```bash
+git submodule update
```

Comment:
You can also just run 

git submodule update --init

in the main repository to have it update all submodules; the --init flag tells it to grab all submodules as necessary.

Reply:
(This is in place of the two steps you mention.)

Reply:
Feel free to change this or not, and then squash and merge.

Reply:
yeah, good idea. I will add it and merge.